apps/server/src/api/user.ts

```diff
@@ -0,0 +1,126 @@
+import { Api } from '@modtree/repo-api'
+import { CustomReqBody, CustomReqQuery } from '@modtree/types'
+import { copy, emptyInit, flatten } from '@modtree/utils'
+import { Request } from 'express'
+
+type ListRequest = {
+  id?: string
+  authZeroId?: string
+  email?: string
+}
+
+export class UserApi {
+  /**
+   * creates a User
+   *
+   * @param {Api} api
+   */
+  static create = (api: Api) => async (req: Request) => {
+    const props = emptyInit.User
+    copy(req.body, props)
+    return api.userRepo.initialize(props).then(flatten.user)
+  }
+
+  /**
+   * gets one User by id
+   *
+   * @param {Api} api
+   */
+  static get = (api: Api) => async (req: CustomReqQuery<ListRequest>) => {
+    return api.userRepo.findOneById(req.params.userId).then(flatten.user)
+  }
+
+  /**
+   * gets one User by primary keys
+   * at least one of id, authZeroId, or email
+   *
+   * @param {Api} api
+   */
+  static getByPrimaryKeys =
+    (api: Api) => async (req: CustomReqBody<ListRequest>) => {
+      const { id, authZeroId, email } = req.body
+      return api.userRepo
+        .findOneOrFail({
+          where: { id, authZeroId, email },
+          relations: api.relations.user,
+        })
+        .then(flatten.user)
+    }
+
+  /**
+   * lists all Users
+   *
+   * @param {Api} api
+   */
+  static list = (api: Api) => async (req: CustomReqQuery<ListRequest>) => {
+    const { id, authZeroId, email } = req.query
```

Comment:
might want to use `req.body` for email

Reply:
get requests don't take body parameters

Reply:
I'm not sure if email works as a query param. Also we might not want to put it as a query param for privacy